src/Compilers/CSharp/Test/Symbol/Symbols/Source/FileModifierTests.cs

```diff
@@ -5004,4 +5004,84 @@ partial class C : I<FI>
         var verifier = CompileAndVerify(new[] { (source0, "F0.cs"), (source1, "F1.cs"), (source2, "F2.cs") }, expectedOutput: "12");
         verifier.VerifyDiagnostics();
     }
+
+    [Fact]
+    public void ShadowNamespace_01()
+    {
+        var source1 = """
+            namespace App.Widget
+            {
+                class Inner { }
+            }
+
+            """;
+
+        var source2 = """
+            namespace App
+            {
+                file class Widget { }
+            }
+
+            """;
+
+        var comp = CreateCompilation(new[] { (source1, "File1.cs"), (source2, "File2.cs") });
+        comp.VerifyDiagnostics();
+
+        comp = CreateCompilation(source1 + source2);
+        comp.VerifyDiagnostics(
+            // (7,16): error CS9071: The namespace 'App' already contains a definition for 'Widget' in this file.
```

Comment:
We want the error in this case because the current file also defines the namespace.

Reply:
Compare with

https://github.com/dotnet/roslyn/blob/3a4d11033c6791e83d1a747c4eb64ba5589f396e/src/Compilers/CSharp/Test/Symbol/Symbols/Source/FileModifierTests.cs#L1977-L1992

Reply:
I didn't fully follow either why we want an error in this scenario. 
I would have thought that would be allowed and within the context of file1 `App.Widget` means the namespace and within file2 it would mean the file-local type.